components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1509,6 +1531,21 @@ private String handleServerErrorAuthorization(OAuthMessage oAuthMessage, OIDCSes
         OAuthProblemException oauthProblemException = OAuthProblemException.error(
                 errorCode, errorMsg);
         authorizationResponseDTO.setError(HttpServletResponse.SC_FOUND, errorMsg, errorCode);
+        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+            DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                    OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, "handle-authorization");
+            if (oauth2Params != null) {
+                diagnosticLogBuilder.inputParam("client id", oauth2Params.getClientId())
+                        .inputParam("application name", oauth2Params.getApplicationName())
+                        .inputParam("redirect uri", oauth2Params.getRedirectURI())
+                        .inputParam("scope", oauth2Params.getScopes())
+                        .inputParam("response type", oauth2Params.getResponseType());
+            }
+            diagnosticLogBuilder.resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                    .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
```

Comment:
the error code defined as SERVER_ERROR? If so, could this log detail level be considered an application level?

Reply:
We can change the log level to internal here. The main reason to define it has an application level is because then the user knows that the issue occurred due to a system failure, not due to any other issues